### PR TITLE
Fe testing

### PR DESCRIPTION
**Pull request by Marek-Kazimierczak**
Unit tests

### CODE REVIEW

**Comment by Marek-Kazimierczak** [comment]
Here are all of my tests I've done till now I would be glad if you can check them and give me some feedback. 
--- work in progress ---

File: frontend/src/__tests__/components/Categoties.test.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    name: "commandline",
+  },
+  {
+    id: 6,
+    name: "crystal",
+  },
+];
+
+describe("<Categories/>", () => {
+  it("renders properly", () => {
```
Comment (MarcinCholewka) [inline frontend/src/__tests__/components/Categoties.test.js:35]: regarding the DRY rule, mock store globally. 
Also use cleanup to unmounts the component from the container and destroys the container.

```suggestion
describe("<Categories/>", () => {
  const mockStore = configureMockStore([]);
  let store;
  
  beforeEach(() => {
    store = mockStore({
      categories,
    });
  });

  afterEach(cleanup);
  
  it("renders properly", () => {
```

File: frontend/src/__tests__/components/Categoties.test.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    const mockStore = configureMockStore([]);
+    const store = mockStore({
+      categories,
+    });
```
Comment (MarcinCholewka) [inline frontend/src/__tests__/components/Categoties.test.js:39]: ✂️

File: frontend/src/__tests__/components/Categoties.test.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    );
+
+    const { asFragment } = render(component);
+
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it("renders empty list when zero categories provided", () => {
+    const mockStore = configureMockStore([]);
+
```
Comment (MarcinCholewka) [inline frontend/src/__tests__/components/Categoties.test.js:54]: ✂️

File: frontend/src/__tests__/components/Categoties.test.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    });
+
+    const component = (
+      <Provider store={store}>
+        <Categories />
+      </Provider>
+    );
+
+    const { queryAllByTestId } = render(component);
+
```
Comment (MarcinCholewka) [inline frontend/src/__tests__/components/Categoties.test.js:66]: ✂️ delete empty line

File: frontend/src/__tests__/components/Categoties.test.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    expect(items.length).toEqual(0);
+  });
+
+  it("renders list with correct names when categories provided", () => {
+    const mockStore = configureMockStore([]);
+
+    const store = mockStore({
+      categories,
+    });
```
Comment (MarcinCholewka) [inline frontend/src/__tests__/components/Categoties.test.js:77]: ✂️

File: frontend/src/__tests__/components/CodeBlock.test.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import CodeBlock from "../../components/CodeBlock";
+
+describe("<CodeBlock/>", () => {
+  it("renders properly", () => {
+    const { asFragment } = render(<CodeBlock />);
```
Comment (MarcinCholewka) [inline frontend/src/__tests__/components/CodeBlock.test.js:7]: add empty line after const

File: frontend/src/__tests__/components/CopyURL.test.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import CopyURL from "../../components/CopyURL";
+
+describe("<CopyURL/>", () => {
+  it("renders properly", () => {
+    const { asFragment } = render(<CopyURL />);
```
Comment (MarcinCholewka) [inline frontend/src/__tests__/components/CopyURL.test.js:7]: add empty line after const

File: frontend/src/__tests__/components/DisplayPost.test.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  const history = createMemoryHistory();
+
+  const component = (
+    <Router history={history}>
+      <DisplayPost />
+    </Router>
+  );
+
+  it("renders correctly", () => {
+    const { asFragment } = render(component);
```
Comment (MarcinCholewka) [inline frontend/src/__tests__/components/DisplayPost.test.js:17]:
```suggestion
    const { asFragment } = component;
```

File: frontend/src/__tests__/components/DisplayPost.test.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it("initializes with `loading` state", () => {
+    const { getByText } = render(component);
```
Comment (MarcinCholewka) [inline frontend/src/__tests__/components/DisplayPost.test.js:23]:
```suggestion
    const { getByText } = component;
```

File: frontend/src/__tests__/components/DisplayPost.test.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { Router } from "react-router-dom";
+import { createMemoryHistory } from "history";
+import { render } from "@testing-library/react";
```
Comment (MarcinCholewka) [inline frontend/src/__tests__/components/DisplayPost.test.js:4]:
```suggestion
import { render, cleanup } from "@testing-library/react";
```

File: frontend/src/__tests__/components/CodeBlock.test.js (hunk 6 above)
Comment (Marek-Kazimierczak) [reply frontend/src/__tests__/components/CodeBlock.test.js:7]: 👍

File: frontend/src/__tests__/components/DisplayPost.test.js (hunk 9 above)
Comment (Marek-Kazimierczak) [reply frontend/src/__tests__/components/DisplayPost.test.js:23]: 알았어 대장님  ✅

File: frontend/src/__tests__/components/DisplayPost.test.js (hunk 10 above)
Comment (Marek-Kazimierczak) [reply frontend/src/__tests__/components/DisplayPost.test.js:4]: 알았지 과장님 👌

File: frontend/src/components/DisplayPost.js [hunk 11]
```diff
@@ -4,7 +4,7 @@ import { useParams } from "react-router-dom";
 import Markdown from "./Markdown";
 import CopyPostURL from "./CopyURL";
 
-const DisplayPost = props => {
+const DisplayPost = (props) => {
```
Comment (MarcinCholewka) [inline frontend/src/components/DisplayPost.js:7]:
```suggestion
const DisplayPost = props => {
```

**Pull request closed without merging**